src/main/kotlin/ink/abb/pogo/scraper/tasks/ReleasePokemon.kt

```diff
@@ -9,62 +9,60 @@
+import java.util.concurrent.atomic.AtomicInteger
 
 class ReleasePokemon : Task {
     override fun run(bot: Bot, ctx: Context, settings: Settings) {
-        val pokemons = ctx.api.cachedInventories.pokebank.pokemons ?: return
+        val pokemonMap = ctx.api.inventory.pokemon
         // prevent concurrent modification exception
-        val groupedPokemon = pokemons.groupBy { it.pokemonId }
+        val groupedPokemon = pokemonMap.map { it.value }.groupBy { it.pokemonData.pokemonId }
         val sortByIV = settings.sortByIv
         val pokemonCounts = hashMapOf<String, Int>()
 
         groupedPokemon.forEach {
             val sorted = if (sortByIV) {
-                it.value.sortedByDescending { it.getIv() }
+                it.value.sortedByDescending { it.pokemonData.getIv() }
             } else {
-                it.value.sortedByDescending { it.cp }
+                it.value.sortedByDescending { it.pokemonData.cp }
             }
             for ((index, pokemon) in sorted.withIndex()) {
                 // don't drop favorited, deployed, or nicknamed pokemon
-                val isFavourite = pokemon.nickname.isNotBlank() || pokemon.isFavorite || !pokemon.deployedFortId.isEmpty()
+                val isFavourite = pokemon.pokemonData.nickname.isNotBlank() || pokemon.pokemonData.favorite != 0 || !pokemon.pokemonData.deployedFortId.isEmpty()
                 if (!isFavourite) {
-                    val ivPercentage = pokemon.getIvPercentage()
+                    val ivPercentage = pokemon.pokemonData.getIvPercentage()
                     // never transfer highest rated Pokemon (except for obligatory transfer)
-                    if (settings.obligatoryTransfer.contains(pokemon.pokemonId) || index >= settings.keepPokemonAmount) {
-                        val (shouldRelease, reason) = pokemon.shouldTransfer(settings, pokemonCounts)
+                    if (settings.obligatoryTransfer.contains(pokemon.pokemonData.pokemonId) || index >= settings.keepPokemonAmount) {
+                        val (shouldRelease, reason) = pokemon.pokemonData.shouldTransfer(settings, pokemonCounts,
+                                bot.api.inventory.candies.getOrPut(PokemonMetaRegistry.getMeta(pokemon.pokemonData.pokemonId).family, { AtomicInteger(0) }))
 
                         if (shouldRelease) {
-                            Log.yellow("Going to transfer ${pokemon.pokemonId.name} with " +
-                                    "CP ${pokemon.cp} and IV $ivPercentage%; reason: $reason")
-                            val result = pokemon.transferPokemon()
+                            Log.yellow("Going to transfer ${pokemon.pokemonData.pokemonId.name} with " +
+                                    "CP ${pokemon.pokemonData.cp} and IV $ivPercentage%; reason: $reason")
+                            bot.api.queueRequest(ink.abb.pogo.api.request.ReleasePokemon().withPokemonId(pokemon.pokemonData.id)).subscribe {
```

Comment:
.toBlocking() needed to avoid doubletransfer? also a successfully transfered log